monsterCatalog.py

```diff
@@ -188,180 +190,143 @@ def monsters():
-        response = flask.make_response(
-            json.dumps("Token's client ID does not match app's."), 401)
-        print "Token's client ID does not match app's."
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    stored_credentials = flask.session.get('credentials')
-    stored_gplus_id = flask.session.get('gplus_id')
-    if stored_credentials is not None and gplus_id == stored_gplus_id:
-        response = flask.make_response(json.dumps('Current user is already connected.'), 200)
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    # Store the access token in the session for later use.
-    flask.session['credentials'] = credentials.access_token
-    flask.session['gplus_id'] = gplus_id
-
-    # Get user info
-    userinfo_url = "https://www.googleapis.com/oauth2/v1/userinfo"
-    params = {'access_token': credentials.access_token, 'alt': 'json'}
-    answer = requests.get(userinfo_url, params=params)
-
-    data = answer.json()
-
-    flask.session['username'] = data['name']
-    flask.session['picture'] = data['picture']
-    flask.session['email'] = data['email']
-
-    output = ''
-    output += '<h1>Welcome, '
-    output += flask.session['username']
-    output += '!</h1>'
-    output += '<img src="'
-    output += flask.session['picture']
-    output += ' " style = "width: 300px; height: 300px;border-radius: 150px;-webkit-border-radius: 150px;-moz-border-radius: 150px;"> '
-    flask.flash("you are now logged in as %s" % flask.session['username'])
-    print "done!"
-    return output
+
+def set_state():
+    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                    for x in xrange(32))
+    flask.session['state'] = state
+    return state
 
 
 @app.route('/fbconnect', methods=['POST'])
 def fbconnect():
-    if flask.request.args.get('state') != flask.session['state']:
-        response = flask.make_response(json.dumps('Invalid state parameter.'), 401)
-        response.headers['Content-Type'] = 'application/json'
-        return response
-    access_token = flask.request.data
-    print "access token received %s " % access_token
+    # Check that local state parameter matches the one sent to Facebook
+    fb_returned_state = flask.request.args.get('state')
+    check_state_parameter(fb_returned_state)
 
+    # Exchange for long-lived access token
+    short_lived_access_token = flask.request.data
+    long_lived_access_token = exchange_fb_token(short_lived_access_token)
+
+    # Get FB user data; if email doesn't exist, add user
+    fb_user_data = get_fb_user_data(long_lived_access_token)
+    user_id = add_or_verify_user(fb_user_data)
+
+    if user_id:
+        print fb_user_data['id']
+        flask.session['facebook_id'] = fb_user_data['id']
+        flask.session['access_token'] = long_lived_access_token
+        print flask.session
+        success_message = 'Login successful.'
```

Comment:
You can just put these messages inline rather than making a variable unless you'll re-use.  Minor though.